Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -436,40 +436,60 @@ public class PBXProjGenerator {
-            allFilePaths += subGroups.filePaths
+            allSourceFiles += subGroups.sourceFiles
             groupChildren.append(subGroups.groups.first!.reference)
             groups += subGroups.groups
         }
 
-        for filePath in filePaths {
-            let fileReference = getFileReference(path: filePath, inPath: path)
-            groupChildren.append(fileReference)
+        // create variant groups of the base localisation first
+        var baseLocalisationVariantGroups = [PBXVariantGroup]()
+        if let baseLocalisedDirectory = localisedDirectories.first(where: { $0.lastComponent == "Base.lproj" }) {
+            for filePath in try baseLocalisedDirectory.children() {
+                let variantGroup = PBXVariantGroup(reference: generateUUID(PBXVariantGroup.self, filePath.lastComponent),
+                                                   children: Set(),
+                                                   name: filePath.lastComponent,
+                                                   sourceTree: .group)
+                addObject(variantGroup)
+                baseLocalisationVariantGroups.append(variantGroup)
+                groupChildren.append(variantGroup.reference)
+
+                let fileReference = variantGroup.reference
+                let buildFile = PBXBuildFile(reference: generateUUID(PBXBuildFile.self, fileReference), fileRef: fileReference, settings: nil)
+                allSourceFiles.append(SourceFile(path: filePath, fileReference: fileReference, buildFile: buildFile))
+            }
         }
 
+        // add references to localised resources into base localisation variant groups
         for localisedDirectory in localisedDirectories {
-            for path in try localisedDirectory.children() {
-                let filePath = "\(localisedDirectory.lastComponent)/\(path.lastComponent)"
-                let fileReference = PBXFileReference(reference: generateUUID(PBXFileReference.self, localisedDirectory.lastComponent), sourceTree: .group, name: localisedDirectory.lastComponentWithoutExtension, path: filePath)
-                addObject(fileReference)
+            let localisationName = localisedDirectory.lastComponentWithoutExtension
 
-                let variantGroup = PBXVariantGroup(reference: generateUUID(PBXVariantGroup.self, path.lastComponent), children: Set([fileReference.reference]), name: path.lastComponent, sourceTree: .group)
-                addObject(variantGroup)
+            for filePath in try localisedDirectory.children() {
 
-                fileReferencesByPath[path] = variantGroup.reference
-                groupChildren.append(variantGroup.reference)
-                allFilePaths.append(path)
+                // find base localisation variant group
+                let name = filePath.lastComponentWithoutExtension
+                guard let variantGroup = baseLocalisationVariantGroups.first(where: { Path($0.name).lastComponentWithoutExtension == name }) else {
```

Comment:
what if a localised file isn't inside a `Base.lproj`? Does this then not add the file reference anymore?

Reply:
Yes It will do not add the reference. It can be a problem in the case that resources placed in en.lproj but not in Base.lproj. For example a case an user removed resource in Base.lproj but not one in en.lproj.
I will fix it.

Reply:
If the project was created in the usual way, it looks like Base.lproj is included.

> The project folder should contain one folder named Base.lproj and other language-specific folders with the .lproj extension. The prefix for the language folders is the language ID, as described in Language and Locale IDs.

https://developer.apple.com/library/content/documentation/MacOSX/Conceptual/BPInternational/MaintaingYourOwnStringsFiles/MaintaingYourOwnStringsFiles.html